dll/win32/shell32/CDefView.cpp

```diff
@@ -2182,6 +2198,15 @@ LRESULT CDefView::OnContextMenu(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL &b
     if (FAILED_UNEXPECTEDLY(hResult))
         return 0;
 
+    if (m_pCommDlgBrowser && !(GetCommDlgViewFlags() & CDB2GVF_NOSELECTVERB))
```

Comment:
```suggestion
    if (m_pCommDlgBrowser.p && !(GetCommDlgViewFlags() & CDB2GVF_NOSELECTVERB))
```
Because that `m_pCommDlgBrowser` is a `CComPtr<>` I wonder whether we really have to use its `.p` member for doing the NULL vs. non-NULL checks.
(Note also that above in this file, `void CDefView::CheckToolbar()` seems to miss the `.p` too.)

Reply:
I don't know if the ROS smart pointer implementation is broken or what the problem is but it certainly is annoying. Sometimes you get `error: no match for 'operator&&' (operand types are 'const ATL::CComPtr<ICommDlgBrowser>' and 'bool')` without the `.p`.